pkg/route/graph/analysis/analysis.go

```diff
@@ -70,6 +76,56 @@ route:
 
 				continue route
 			}
+
+			if routeNode.Spec.Port == nil {
+				continue route
+			}
+
+			routePortString := routeNode.Spec.Port.TargetPort.String()
+			if routePort, err := strconv.Atoi(routePortString); err == nil {
+				for _, port := range svcNode.Spec.Ports {
+					if port.TargetPort.IntValue() == routePort {
+						continue route
+					}
+				}
+
+				// route has numeric port, service has no port with that number as a targetPort.
+				marker := osgraph.Marker{
+					Node:         routeNode,
+					RelatedNodes: []graph.Node{svcNode},
+
+					Severity: osgraph.WarningSeverity,
+					Key:      WrongRoutePortWarning,
+					Message: fmt.Sprintf("%s has a port specified (%d) but %s has no such targetPort.",
+						f.ResourceName(routeNode), routePort, f.ResourceName(svcNode)),
+				}
+
+				if len(svcNode.Spec.Ports) == 1 {
+					marker.Suggestion = osgraph.Suggestion(fmt.Sprintf("oc patch %s -p '{\"spec\":{\"port\":{\"targetPort\": %d}}}'", f.ResourceName(routeNode), svcNode.Spec.Ports[0].TargetPort.IntValue()))
+				}
+				markers = append(markers, marker)
+
+				continue route
+			}
+
+			for _, port := range svcNode.Spec.Ports {
+				if port.Name == routePortString {
+					continue route
+				}
+			}
+
+			// route has named targetPort, service has no port with that name.
+			markers = append(markers, osgraph.Marker{
```

Comment:
Can't you do the same logic for a suggestion that you used above?  If you have a service with exactly one port, they could use a patch.  I liked that.